### PR TITLE
resolve security concern with unsigned container images

### PR DESCRIPTION
**Pull request by arthurus-rex**
Follow-up to a security advisory I raised 3 weeks ago about unsigned images within Kepler workflow.  In this PR, I have introduced cosign into the image-building workflow (specifically in `image.yml`).  I tested it successfully on my own registry and then adjusted the URLs.

### CODE REVIEW

File: .github/workflows/image.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ on: # yamllint disable-line rule:truthy
 jobs:
   image_build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
```
Comment (sthaha) [inline .github/workflows/image.yml:24]: mind adding a note why these are needed?

**Review by sthaha — CHANGES_REQUESTED** [review]
Thanks for the patch @arthurus-rex . I am requesting change to rebase the required change on top of the latest main.

File: .github/workflows/image.yml (hunk 1 above)
Comment (arthurus-rex) [reply .github/workflows/image.yml:24]: Yes, the id-token must be set to write in order for the OIDC to be used during the signing.  And, in GitHub Actions, when any permissions are specified, all others are set to none.  So, the other two permissions allow this action to read/write from registry.

**Comment by arthurus-rex** [comment]
I had a serious issue with the rebasing, I will be making a new pull request shortly.

**Pull request closed without merging**